grid_map_core/src/GridMap.cpp

```diff
@@ -65,6 +67,49 @@ void GridMap::setGeometry(const SubmapGeometry& geometry) {
   setGeometry(geometry.getLength(), geometry.getResolution(), geometry.getPosition());
 }
 
+void GridMap::grow(const Length& length, const Direction direction, float value)
+{
+  if (length(0) < length_(0) || length(1) < length_(1))
+  {
+    return;
+  }
+
+  Length delta(length(0) - length_(0), length(1) - length_(1));
+
+  Size size;
+  size(0) = static_cast<int>(round(length(0) / resolution_));
+  size(1) = static_cast<int>(round(length(1) / resolution_));
+  conservativeResize(size, direction, value);
+  length_ = length;
```

Comment:
I had to replace this line by this one to avoid alignment issues:

```suggestion
  length_ = (size_.cast<double>() * resolution_).matrix();
```

Reply:
Taken from here:
https://github.com/ANYbotics/grid_map/blob/c23dcde8f0fc5ead0a30d24179344d71a7eba163/grid_map_core/src/GridMap.cpp#L57